esmf_regrid/tests/unit/schemes/test_regrid_rectilinear_to_rectilinear.py

```diff
@@ -149,3 +150,19 @@ def _add_metadata(cube):
     # Check metadata and coords.
     result.data = expected_data
     assert expected_cube == result
+
+
+def test_laziness():
+    """Test that regridding is lazy when source data is lazy."""
+    n_lons = 6
+    n_lats = 5
+    lon_bounds = (-180, 180)
+    lat_bounds = (-90, 90)
+    src = _grid_cube(n_lons, n_lats, lon_bounds, lat_bounds, circular=True)
+    tgt = _grid_cube(n_lons, n_lats, lon_bounds, lat_bounds, circular=True)
+
+    src_data = da.ones([n_lats, n_lons])
+    src.data = src_data
+    assert src.has_lazy_data()
+    result = regrid_rectilinear_to_rectilinear(src, tgt)
+    assert result.has_lazy_data()
```

Comment:
Do we have a test that ensures that lazy regridding returns expected results after computing?

Reply:
Could be a parameterised test, e.g. pass in the same data, lazy and non lazy, and ensure they both give the same reference result

Reply:
I've added a bit to flesh out the test.

Reply:
So what are you doing here? regridding to itself and ensuring no data has changed?

Reply:
Effectively, yes. With chunking added in all directions. This effectively makes sure that the dimensions aren't mixed up and everything stays in the correct order. I feel like this should cover most of the obvious bugs you might expect from dealing with chunked data.

Reply:
Though with that said, there are less obvious bugs such as this one https://github.com/dask/dask/issues/6907 which I imagine would not be picked up. This particular bug seems to be obscure enough that we probably wouldn't pick it up even with a more sophisticated test.

Reply:
Agree, seems reasonable.  We should keep track of #61 on the backlog though and make sure we get to it at some point as we can use a slightly more complex case there to ensure things are working as expected.

Reply:
I suppose it could be an idea to have a regridding between different grids, but to compare the result from realised regridding to ensure it matches the result from lazy regridding. That way the test is closer to actual regridding use cases.